core/src/main/java/org/apache/iceberg/hadoop/HadoopTableOperations.java

```diff
@@ -353,8 +346,12 @@ int findVersion() {
    * @param src the source file
    * @param dst the destination file
    */
-  private void renameToFinal(FileSystem fs, Path src, Path dst) {
+  private void renameToFinal(FileSystem fs, Path src, Path dst, int nextVersion) {
     try {
+      lockManager.acquire(dst.toString(), Thread.currentThread().getName());
+      if (fs.exists(dst)) {
+        throw new CommitFailedException("Version %d already exists: %s", nextVersion, dst);
+      }
```

Comment:
Nit: please add a newline between control flow blocks and the following statement.

Reply:
done